lisa/transformers/kernel_source_installer.py

```diff
@@ -325,6 +315,23 @@ def _install_build_tools(self, node: Node, build_deps: list[str]) -> None:
                     os.install_packages(package)
             os.group_install_packages("Development Tools")
 
+            # if the kernel requires devtoolset, install its gcc
+            has_devtoolset = [
+                    pkg
+                    for pkg in build_deps
+                    if pkg.startswith("devtoolset")
+            ]
+            if has_devtoolset:
+                assert len(has_devtoolset) == 1, "only one devtoolset can be given"
+                devtoolset = has_devtoolset[0]
+                node.os.install_packages([devtoolset])
```

Comment:
Use `node.os.install_packages(devtoolsets)`, and it doesn't need to take 1st from previous line. After assertion, the array is guaranteed 1 item.